layer/tracker/ui_progress_tracker.py

```diff
@@ -62,11 +59,12 @@ def track(self) -> Iterator["RunProgressTracker"]:
             self._progress.refresh()
 
     def _get_url(self, asset_type: AssetType, name: str) -> URL:
+        project_full_name = context.get_project_full_name()
```

Comment:
same comment here: is there no way to pass one thing explicitly? if passing `project_full_name` does not make sense anymore, maybe context does?

it could be that it is the best way, Im just very cautious about `context` because it hides dependencies and side effects a lot. 

At least let's use it implicitly only for reading like here, not for mutating it

Reply:
agreed, I think we should pass in the RunContext here.